### ISSUE TITLE
UnusedVariable false positive with thrown and caught exceptions

### ISSUE THREAD

**Issue by Daeroni** [vimeo/psalm#5700]
If a variable is modified just before an exception is thrown, and then caught in the same context and the variable used, psalm emits an UnusedVariable issue.

https://psalm.dev/r/95027a9b22
https://3v4l.org/9UkSq

**Comment by muglug** [comment]
Also a different false-positive for type assignment (because unused variable tracking does both): https://psalm.dev/r/621191710b

**Comment by tostiheld** [comment]
I also ran in this issue, and the issue went away after I downgraded all the way to 4.6.4

**Comment by thepercival** [comment]
I also have a same sort of problem. In the catch I call the same function.
https://psalm.dev/r/43e4ab49f4

**Comment by orklah** [comment]
@thepercival your case seems discutable because `$i++` is equivalent to `$i = $i + 1`, so it makes the addition then an assignment. You do not need the assignment for this:
https://psalm.dev/r/f747b16b76
and if you add the assignment, the result is the same:
https://psalm.dev/r/f6d3f5ba92

**Comment by thepercival** [comment]
You are right. I did not realize that (++$var) was an implicit assignment. Sorry for the comment. I fixed it with a simple addition.

**Comment by iwiznia** [comment]
Encountered [this](https://github.com/vimeo/psalm/issues/5700#issuecomment-830827361) exact case, can we expect a resolution of this at some point? Or can someone point me in the right direction so I can try to send a PR to solve this?
